### PR TITLE
Add Header Documentation

### PR DESCRIPTION
**Pull request by melissatdiamond**
Add header documentation now that there are more header properties (#3888).

### CODE REVIEW

**Comment by melissatdiamond** [comment]
Add header.md: 
screen shot 2018-06-26 at 3 47 30 pm

Remove Header info from facet.md:
screen shot 2018-06-26 at 3 46 51 pm

Add header to config.md:
screen shot 2018-06-26 at 3 52 54 pm

**Comment by domoritz** [comment]
The last table looks empty.

File: examples/specs/facet_custom.vl.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (kanitw) [inline examples/specs/facet_custom.vl.json]: Should we name it `facet_custom_header`?

File: site/docs/encoding/header.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+---
+layout: docs
+menu: docs
+title: Header
+permalink: /docs/header.html
+---
+
+Headers provide a title and labels for [faceted Trellis plots](facet.html). A header's title describes the plot in sum, while a header's labels describe the subplots individually.
```
Comment (kanitw) [inline site/docs/encoding/header.md]: > faceted Trellis plots

Comment (kanitw) [reply site/docs/encoding/header.md]: > A header's title describes the plot in sum, while a header's labels describe the subplots individually.

File: site/docs/encoding/header.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+By default, Vega-Lite automatically creates axes with default properties for `row` and `column` channels of a faceted view.
```
Comment (kanitw) [inline site/docs/encoding/header.md]: creates header*

File: examples/specs/facet_custom.vl.json (hunk 1 above)
Comment (kanitw) [reply examples/specs/facet_custom.vl.json]: Rationale: we can customize other things in facet too (e.g., spacing between facet) this example focuses on customizing header.

**Pull request merged**